Basic-Car-Maintenance/Shared/Dashboard/ViewModels/DashboardViewModel.swift

```diff
@@ -64,5 +64,9 @@ class DashboardViewModel: ObservableObject {
             .collection("maintenance_events")
             .document(documentId)
             .delete()
+
+        if let index = events.firstIndex(where: {$0.id == documentId}) {
+            events.remove(at: index)
+        }
```

Comment:
This actually doesn't need to be fixed, because for 1, it's out of the scope of the issue you were working on (so be sure to stay on task), but 2nd because the button role is `.destructive` where this is used on the `DashboardView` I believe this happens automatically

I'm going to merge your PR, but remove this, because it's out of scope for this issue

If you think this is still an issue, file a bug to work on!

Reply:
I made this change because the ```StateObject viewmodel``` in ```DashboardView``` is not updating the UI until we update the events object.